### DISCUSSION TITLE
"Zigbee Home" DIY firmware project for nRF52840 aim to provide similar functionality to ESPHome

### DISCUSSION THREAD

**Discussion by Hedda** [rbaron/b-parasite#178] [General]
FYI, a new project by [ffenix113](https://github.com/ffenix113) aims to (eventually) provide similar functionality to [ESPHome](https://esphome.io/) but for nRF52840 based Zigbee devices: 

https://community.home-assistant.io/t/zigbee-home-diy-firmware-project-aim-to-provide-similar-functionality-to-esphome/675865

The goal is to eventually enable any user to make their own Zigbee devices using only YAML without any need to touch the source code.

This DIY firmware is initially designed on nRF52840 it is still in very early development so good idea to post any early feedback/input now:

https://github.com/ffenix113/zigbee_home

https://ffenix113.github.io/zigbee_home/

Other than following the ESPHome concept it also seems to take inspiration from PVTO (Zigbee Switch Configurable Firmware) which also allows users to make DIY Zigbee  devices but is closed source:

https://ptvo.info/zigbee-switch-configurable-firmware-router-199/

PS: [ffenix113](https://github.com/ffenix113) has also upstreamed patches to the Zigbee2MQTT project so it can support devices that can change features dynamically.

PPS: Home Assistant's ZHA integration might need similar if using custom Zigbee clusters and attributes, see -> https://github.com/zigpy/zigpy/discussions/1312

**Comment by Hedda** [comment]
FYI, [ffenix113](https://github.com/ffenix113) has now added ADC (Analog To Digital Sensor) + support for required soil moisture sensors to his "Zigbee Home" project, see:

* https://github.com/ffenix113/zigbee_home/discussions/5

**Reply by rbaron** [reply]
Thanks for the heads up @Hedda.
